packages/tokens/src/rwa/identity_storage_registry/storage.rs

```diff
@@ -0,0 +1,502 @@
+    }
+    e.storage().persistent().set(&count_key, &count);
+}
+
+/// Modifies an existing country profile by its index.
+///
+/// # Arguments
+///
+/// * `e` - The Soroban environment.
+/// * `account` - The account address whose profile is being modified.
+/// * `index` - The index of the profile to modify.
+/// * `profile` - The new country profile data.
+///
+/// # Errors
+///
+/// * [`IRSError::CountryProfileNotFound`] - If the index is out of bounds.
+///
+/// # Events
+///
+/// * topics - `["country_modified", account: Address, profile: CountryProfile]`
+/// * data - `[]`
+///
+/// # Security Warning
+///
+/// **IMPORTANT**: This function bypasses authorization checks and should only
+/// be used:
+/// - During contract initialization/construction
+/// - In admin functions that implement their own authorization logic
+///
+/// Using this function in public-facing methods may create significant security
+/// risks as it could allow unauthorized modifications.
+pub fn modify_country_profile(e: &Env, account: &Address, index: u32, profile: &CountryProfile) {
+    let key = IRSStorageKey::CPEnumerable(CPEnumerableKey { account: account.clone(), index });
+
+    if !e.storage().persistent().has(&key) {
+        panic_with_error!(e, IRSError::CountryProfileNotFound)
+    }
+    e.storage().persistent().set(&key, profile);
+
+    emit_country_profile_event(e, CountryProfileEvent::Modified, account, profile);
+}
+
+/// Deletes a country profile by its index. This operation that swaps the
+/// profile to be deleted with the last profile in the list.
+///
+/// # Arguments
+///
+/// * `e` - The Soroban environment.
+/// * `account` - The account address whose profile is being deleted.
+/// * `index` - The index of the profile to delete.
+///
+/// # Errors
+///
+/// * [`IRSError::CountryProfileNotFound`] - If the index is out of bounds.
+/// * [`IRSError::EmptyCountryProfiles`] - If deleting the last profile is
+///   attempted.
+///
+/// # Events
+///
+/// * topics - `["country_removed", account: Address, profile: CountryProfile]`
+/// * data - `[]`
+///
+/// # Security Warning
+///
+/// **IMPORTANT**: This function bypasses authorization checks and should only
+/// be used:
+/// - During contract initialization/construction
+/// - In admin functions that implement their own authorization logic
+///
+/// Using this function in public-facing methods may create significant security
+/// risks as it could allow unauthorized modifications.
+pub fn delete_country_profile(e: &Env, account: &Address, index: u32) {
+    let current_key =
+        IRSStorageKey::CPEnumerable(CPEnumerableKey { account: account.clone(), index });
+    let profile_to_remove = e
+        .storage()
+        .persistent()
+        .get(&current_key)
+        .unwrap_or_else(|| panic_with_error!(e, IRSError::CountryProfileNotFound));
+
+    let count = get_country_profile_count(e, account);
+    // Can't overflow because `profile_to_remove` would panic if count == 0
```

Comment:
did you mean `get_country_profile_count()` would panic if `count==0` instead of `profile_to_remove`?